lib/pc_schema.c

```diff
@@ -635,3 +635,36 @@ pc_schema_get_size(const PCSCHEMA *s)
 {
 	return s->size;
 }
+
+/**
+* Return true if s1 and s2 have the same dimensions at the same
+* positions, otherwise return false.
+*/
+uint32_t
+pc_schema_similar(const PCSCHEMA *s1, const PCSCHEMA *s2)
+{
+	size_t i;
+
+	if ( s1->ndims != s2->ndims )
+		return PC_FALSE;
+
+	for ( i = 0; i < s1->ndims; i++ )
+	{
+		PCDIMENSION *s1dim = s1->dims[i];
+		PCDIMENSION *s2dim = s2->dims[i];
+
+		if ( strcmp(s1dim->name, s2dim->name) != 0 )
+			return PC_FALSE;
+
+		if ( s1dim->interpretation != s2dim->interpretation )
+			return PC_FALSE;
+
+		if ( s1dim->scale != s2dim->scale )
+			return PC_FALSE;
+
+		if ( s1dim->offset != s2dim->offset )
+			return PC_FALSE;
```

Comment:
I implemented `PC_SetPCId(p pcpatch, pcid int4)` and `PC_SetPCId(p pcpatch, pcid int4, defaults float8)` as discussed. So with my last commit `PC_SetPCId` never re-interprets the patch data.

@mbredif in `pc_patch_set_schema` I've used a simple implementation where each dimension is `memcpy`'ed individually. I know that `memcpy`'s could combined, but I did not want to go that far for now.